docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,253 @@
+
+## Integration-test framework
+
+A typical integration-test framework has three parts:
+
+1. A series of executables with arguments that work together and generate outputs.
+2. A series of expected outputs that should match the output of the executables.
+3. A launcher that starts the tests, compares the outputs to the expected outputs, and determines if the test passes.
+
+In Autoware, we use the [launch_testing](https://github.com/ros2/launch/tree/master/launch_testing) framework.
+
+### Smoke tests
+
+Autoware has a dedicated API for smoke testing. To use this framework, in `package.xml` add:
+
+```xml
+<test_depend>autoware_testing</test_depend>
+```
+
+And in `CMakeLists.txt` add:
+
+```cmake
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  find_package(autoware_testing REQUIRED)
+  add_smoke_test(${PROJECT_NAME} ${NODE_NAME})
+endif()
+```
+
+Doing so adds smoke tests that ensure that a node can be:
+
+1. Launched with a default parameter file.
+2. Terminated with a standard `SIGTERM` signal.
+
+For the full API documentation, refer to the [package design page](https://github.com/autowarefoundation/autoware.universe/blob/main/common/autoware_testing/design/autoware_testing-design.md).
+
+!!! note
+
+    This API is not suitable for all smoke test cases. 
+    It cannot be used when a specific file location (eg: for a map) is required to be passed to the node, or if some preparation needs to be conducted before node launch.
+    In such cases use the manual solution from the [component test section below](#integration-test-with-a-single-node-component-test).
+
+### Integration test with a single node: component test
+
+The simplest scenario is a single node. In this case, the integration test is commonly referred to as a component test.
+
+To add a component test to an existing node, follow the example of the `lanelet2_map_provider` package that has an executable named `lanelet2_map_provider_exe`.
+
+In `package.xml`, add:
+
+```xml
+<test_depend>ros_testing</test_depend>
+```
+
+In `CMakeLists.txt`, add or modify the `BUILD_TESTING` section:
+
+```cmake
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  add_ros_test(
+    test/lanelet2_map_provider_launch.test.py
+    TIMEOUT "30"
+  )
+endif()
+```
+
+!!! note
+
+    The `TIMEOUT` argument is given in seconds; see the [add_ros_test.cmake file](https://github.com/ros2/ros_testing/blob/master/ros_testing/cmake/add_ros_test.cmake) for details.
+
+To create a test, either read the [launch_testing quick-start example](https://github.com/ros2/launch/tree/master/launch_testing#quick-start-example), or follow the steps below.
+
+Taking `test/lanelet2_map_provider_launch.test.py` as an example, first dependencies are imported:
```

Comment:
As above, if this is an actual file, then we should provide a link to it rather than make readers look for it.